### DISCUSSION TITLE
How stable is this as of March-2024

### DISCUSSION THREAD

**Discussion by joekaiser** [vincent-herlemont/native_db#113] [Q&A] [answered]
How stable would you say this is? I'm learning Rust by writing a specific app I want. I'm hopeful this is the embedded DB I'm looking for, but I don't want to have to deal with instabilities on top of learning a new language.

I don't need this to "production" ready, but would you say it is stable enough to use on personal projects without fighting the api, too many breaking changes, or losing data?

**Comment by vincent-herlemont** [comment, marked as answer]
@joekaiser Regarding data, there are no concerns on that front; the data will not be lost. The data layer used by `native_db` (`redb`) is stable. 

However, make sure to fix the database version in our `Cargo.toml`, so you can be sure there will be no regressions.
